src/js/stores/document.js

```diff
@@ -428,14 +429,52 @@ define(function (require, exports, module) {
          * Update layer models when groups are expanded or collapsed.
          *
          * @private
-         * @param {{documentID: number, layerIDs: Immutable.Iterable.<number>, expanded: boolean}} payload
+         * @param {object} payload
+         * @param {number} payload.documentID
+         * @param {Immutable.Iterable.<number>} payload.layerIDs
+         * @param {boolean} payload.expanded
+         * @param {Array.<number>} payload.selected Layer IDs that are newly selected
+         * @param {Array.<number>} payload.deselected Layer IDs that are newly deselected
          */
         _handleGroupExpansion: function (payload) {
             var documentID = payload.documentID,
                 layerIDs = payload.layerIDs,
-                expanded = payload.expanded;
+                expanded = payload.expanded,
+                selectedIDs = payload.selected,
+                deselectedIDs = payload.deselected,
+                suppressChange = true;
+
+            // Suppress the first change event if there will be a second one later
+            if (selectedIDs.length === 0 && deselectedIDs.length === 0) {
+                suppressChange = false;
+            }
+
+            // If there will be a selection change, suppress change event for
+            // initial group expansion change.
+            this._updateLayerProperties(documentID, layerIDs, { expanded: expanded },
+                suppressChange);
+
+            if (!suppressChange) {
+                // A change event has already been emitted, and there are no
+                // further selection changes.
+                return;
```

Comment:
We really need a store-level mechanism to suppress events while multiple changes are happening. But, not for 16.1!